base/reduce.jl

```diff
@@ -642,21 +642,35 @@ end
 
 """
     count(p, itr) -> Integer
+    count(itr) -> Integer
 
 Count the number of elements in `itr` for which predicate `p` returns `true`.
+If `p` is omitted, counts the number of `true` elements in `itr` (which
+should be a collection of boolean values).
 
 ```jldoctest
 julia> count(i->(4<=i<=6), [2,3,4,5,6])
 3
+
+julia> count([true, false, true, true])
+3
 ```
 """
 function count(pred, itr)
     n = 0
     for x in itr
-        n += pred(x)
+        n += pred(x)::Bool
+    end
+    return n
+end
+function count(pred, a::AbstractArray)
+    n = 0
+    for i in eachindex(a)
+        @inbounds n += pred(a[i])::Bool
```

Comment:
Maybe another PR should go through and fix occurrences of `@inbounds` in Base for `AbstractArray`?

It would be nice to have a way to turn this on for `Array` without requiring two methods.  See also #15291